http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/LeoAppServiceInterp.scala

```diff
@@ -249,23 +252,26 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
             petSA,
             appTypeToFormattedByType(req.appType),
             authProvider,
-            config.leoKubernetesConfig.diskConfig
+            config.leoKubernetesConfig.diskConfig,
+            parentWorkspaceId
           )
         )
         lastUsedApp <- getLastUsedAppForDisk(req, diskResultOpt)
+
         saveApp <- F.fromEither(
-          getSavableApp(cloudContext,
-                        appName,
-                        originatingUserEmail,
-                        samResourceId,
-                        req,
-                        diskResultOpt.map(_.disk),
-                        lastUsedApp,
-                        petSA,
-                        nodepool.id,
-                        req.workspaceId,
```

Comment:
Note v1 apps actually had `workspaceId` in the request json. However not all clients were sending it (seems like AoU "allowed" apps were populating it, but not Galaxy). Also it wasn't validated in Leo, so theoretically a caller could provide any workspace ID. For those reasons I'm switching it to the workspace ID retrieved from the google project (which is validated).